9.5.4 Release (#4433)
cypress-io/cypress-documentation: content/_changelogs/9.5.4.md

## 9.5.4

_Released 4/11/2022_

**Bugfixes:**

- Updates were made to silence Electron warnings related to being unable to
  connect to dbus. These errors are normal and expected, and do not result in
  test failures. Because they are always present when running Electron inside
  docker containers, it has incorrectly led people to believe it is the
  root-cause of an error within their test run. By silencing these errors, it
  will improve the debug experience to allow users to focus on meaningful
  warning and error messages. Fixed
  [#19299](https://github.com/cypress-io/cypress/issues/19299).

> - Updates were made to ensure that all `*.enabled` events are sent if Cypress

Reviewer: ```suggestion
- Updates were made to ensure that all `*.enable` events are sent if Cypress
```

  becomes disconnected from the Chrome DevTools Protocol and must re-establish a
  connection to ensure all command logs are displayed to the user. This fixes an
  issue where some command logs, like download or network events, are missing
  once Cypress has disconnected and then reconnected to the Chrome DevTools
  Protocol. Fixed [#20618](https://github.com/cypress-io/cypress/issues/20618).

> - Fixed in issue where `cy.type('{enter}')` was not sending the Enter key for

Reviewer 2: ```suggestion
- Fixed an issue where `cy.type('{enter}')` was not sending the Enter key for
```

  Firefox `v98+`. This was not an issue with Firefox `v97` and below. Fixed
  [#20562](https://github.com/cypress-io/cypress/issues/20562).
- Fixed a regression in [9.3.0](/guides/references/changelog#9-3-0) where
  updates made to split glob patterns provided to the `--spec` CLI parameter on
  commas was incorrectly splitting the patterns. Fixes
  [#20794](https://github.com/cypress-io/cypress/issues/20794).

> - Fixed an issue with `cy.root` to respect the

Reviewer 2: ```suggestion
- Fixed an issue with `cy.root()` to respect the
```

  [`timeout`](/api/commands/root#Arguments) option passed to the command.
  Previously, when the `timeout` option was provided, it was ignored and the
  default timeout was used. Fixed
  [#19985](https://github.com/cypress-io/cypress/issues/19985).
- Updates were made to decrease the length of the Cypress cache path for Windows

>   to ensure installing pre-release versions of the Cypress binary are within the

Reviewer 2: Can we link to the pre-release section of our guides for this so people understand this is not a normal change to our released version?

Author: Good call.

Author: [`a3b549e` (#4433)](https://github.com/cypress-io/cypress-documentation/pull/4433/commits/a3b549e7b7790ee77d3a78b1493745af94410c0c#diff-abc1fe4f8ecd155729980c489195b8ee925e069abb542177b52522016809d299R35)

>   maximum path length of 260 characters. This should improve the experience for
>   Windows users since it's been observed the path length has exceed the default
>   maximum, which resulted in install failures. Fixed in
>   [#20961](https://github.com/cypress-io/cypress/pulls/20961).

Reviewer 2: I wouldn't mention this, it makes it sound like an improvement for all Windows users, this only affects pre-release installs. 

```suggestion
  maximum path length of 260 characters. Fixed in
  [#20961](https://github.com/cypress-io/cypress/pulls/20961).
```

> - Fixed a regression in [9.2.0](/guides/references/changelog#9-2-0) which would
>   sometimes throw an expected error on navigation with `cy.back()` and
>   `cy.go()`. The changes in `9.2.0` were made to mitigate the Chrome ` v97+`
>   change which always triggered an extra load event when an iframe navigated due
>   to a url hashchange. This change was made to maintain how Cypress handles
>   navigation. Fixed [#19749](https://github.com/cypress-io/cypress/issues/19749)

Reviewer 2: Seems unnecessary info that could confused users as it shouldn't affect them. 

```suggestion
- Fixed a regression in [9.2.0](/guides/references/changelog#9-2-0) which would
  sometimes throw an expected error on navigation with `cy.back()` and
  `cy.go()`. Fixed [#19749](https://github.com/cypress-io/cypress/issues/19749)
```

Author: simplified: [`a3b549e` (#4433)](https://github.com/cypress-io/cypress-documentation/pull/4433/commits/a3b549e7b7790ee77d3a78b1493745af94410c0c#diff-abc1fe4f8ecd155729980c489195b8ee925e069abb542177b52522016809d299R40)

  and [#20539](https://github.com/cypress-io/cypress/issues/20539).

> - Corrected the typescript type for a cookie which was incorrectly types as

Reviewer 2: ```suggestion
- Corrected the TypeScript type for a cookie which was incorrectly typed as
```

  `any` when the correct type is `Cookie`. Fixed in
  [#20513](https://github.com/cypress-io/cypress/pull/20513).

> - Add the missing `Cypress.Command.addAll()` typescript types. Fixed

Reviewer 2: ```suggestion
- Add the missing `Cypress.Command.addAll()` TypeScript types. Fixed
```

  [#18886](https://github.com/cypress-io/cypress/issue/18886).

> - Fixed an uncommon error observed in `cy.session` where an error was thrown

Reviewer 2: ```suggestion
- Fixed an uncommon error observed in `cy.session()` where an error was thrown
```

  when no cookies had been set for the session and the user clicks the session

>   command log to view additional details in the dev tools console. Fixed in

Reviewer 2: ```suggestion
  command log to view additional details in the DevTools console. Fixed in
```

  [installing pre-release versions](/guides/getting-started/installing-cypress#Install-pre-release-version)
  of the Cypress binary are within the maximum path length of 260 characters.
  Fixed in [#20961](https://github.com/cypress-io/cypress/pulls/20961).

> - Fixed a regression in [8.6.0](/guides/references/changelog#8-6-0) to which

Reviewer 2: ```suggestion
- Fixed a regression in [8.6.0](/guides/references/changelog#8-6-0) which
```

>   prevented `cy.paused()` from correctly executing when passing the

Reviewer 2: Wrong command name

```suggestion
  prevented `.pause()` from correctly executing when passing the
```
